nservicebus/vscode.md

```diff
@@ -0,0 +1,263 @@
+    "stopAtEntry": false,
+    "console": "externalTerminal"
+}
+```
+
+Most of the values can be considered boilerplate except for the `program` and `cwd` values which should be adjusted for each project:
+
+* `name`: Provides a name that appears for the project in VSCode's Debug dropdown.
+* `type`: The value `coreclr` identifies the plugin that should handle debugging responsibilities.
+* `request`: The value `launch` launches the process, rather than attaching to an existing process.
+* `preLaunchTask`: Indicates a task label from the **tasks.json** file. In this case, it ensures the solution is compiled before trying to launch it.
+* `program`: Identifies the DLL to use as the entry point.
+* `args`: Specifies any command-line arguments that are needed.
+* `cwd`: Specifies the current working directory for the launched `dotnet` process.
+* `stopAtEntry`: If set to true, the debugger will break at the entry point even without a breakpoint set.
+* `console`: Set to `externalTerminal` to use external console windows rather than VSCode's built-in terminal. This is useful when running multiple NServiceBus endpoints.
+
+### Web application
+
+For web application projects that need to launch a browser window the configuration is a little more involved:
+
+```json
+{
+    "name": "EShop.UI",
+    "type": "coreclr",
+    "request": "launch",
+    "preLaunchTask": "build",
+    "program": "${workspaceFolder}/path/to/Website/bin/Debug/netcoreapp2.1/Website.dll",
+    "args": [],
+    "cwd": "${workspaceFolder}/path/to/Website",
+    "stopAtEntry": false,
+    "internalConsoleOptions": "openOnSessionStart",
+    "launchBrowser": {
+        "enabled": true,
+        "args": "${auto-detect-url}",
+        "windows": {
+            "command": "cmd.exe",
+            "args": "/C start ${auto-detect-url}"
+        },
+        "osx": {
+            "command": "open"
+        },
+        "linux": {
+            "command": "xdg-open"
+        }
+    },
+    "env": {
+        "ASPNETCORE_ENVIRONMENT": "Development"
+    },
+    "sourceFileMap": {
+        "/Views": "${workspaceFolder}/path/to/Website/Views"
+    }
+}
+```
+
+* `internalConsoleOptions`: Assuming a single website project in a solution can use VSCode's integrated console, this option controls how it behaves. It's also possible to use the `console` property to use an external console.
+* `launchBrowser`: Contains instructions for launching a browser window on a platform-specific basis.
+* `env`: Additional environment variables to pass to the application, including the common `ASPNETCORE_ENVIRONMENT` variable that affects how an ASP.NET Core web application will read configuration values.
+* `sourceFileMap`: Additional source file mappings passed to the debugger, useful in this case to let the debugger know the path to MVC views.
+
+### Attaching to a process
+
+it's also possible to start debugging from a terminal and then attach to the process with this configuration:
```

Comment:
It's

Reply:
What do you mean by "start debugging from a terminal" ?

Shouldn't that say something like "start a process" since this just need an already running process?